cloud_images/centos7/install_prereqs.sh

```diff
@@ -99,6 +99,41 @@ echo ">>> Remove history"
 unset HISTFILE
 rm -rf /home/*/.*history /root/.*history
 
+echo ">>> Update /etc/hosts on boot"
+update_hosts_script=/usr/local/sbin/dcos-update-etc-hosts
+update_hosts_unit=/etc/systemd/system/dcos-update-etc-hosts.service
+
+mkdir -p "$(dirname $update_hosts_script)"
+
+cat << 'EOF' > "$update_hosts_script"
+#!/bin/bash
+export PATH=/opt/mesosphere/bin:/sbin:/bin:/usr/sbin:/usr/bin
+curl="curl -s -f -m 30 --retry 3"
+fqdn=$($curl http://169.254.169.254/latest/meta-data/local-hostname)
+ip=$($curl http://169.254.169.254/latest/meta-data/local-ipv4)
+echo "Adding $fqdn if $ip is not in /etc/hosts"
+grep ^$ip /etc/hosts > /dev/null || echo -e "$ip\t$fqdn ${fqdn%%.*}" >> /etc/hosts
```

Comment:
There is a one to one mapping between hostnames and IP addresses on AWS, so even if the IP/Hostname changes this will not result in bad data in /etc/hosts.